uxarray/core/zonal.py

```diff
@@ -60,3 +67,243 @@ def _compute_non_conservative_zonal_mean(uxda, latitudes, use_robust_weights=Fal
         result[tuple(idx)] = weighted
 
     return result
+
+
+@njit(cache=True)
+def _sort_points_by_angle(points):
+    """Sort points by longitude angle for proper polygon formation.
+
+    Parameters
+    ----------
+    points : ndarray
+        Array of 3D points on unit sphere, shape (n_points, 3)
+
+    Returns
+    -------
+    ndarray
+        Points sorted by longitude angle
+    """
+    n_points = points.shape[0]
+    if n_points <= 1:
+        return points.copy()
+
+    # Calculate angles (longitude)
+    angles = np.empty(n_points, dtype=np.float64)
+    for i in range(n_points):
+        angles[i] = np.arctan2(points[i, 1], points[i, 0])
```

Comment:
When trying to calculate angles, always try to use the following robust way that incoporate the Kahan's angle formulation to avoid potential floating point precision issues.

https://github.com/UXARRAY/uxarray/blob/484bc2e82813c61c22c4f287d1b257ee7ecf8863/uxarray/grid/utils.py#L9